File: R/layoutWidget.R
#' Layout Editor Widget
#'
#' @param nodes list of node objects
#' @param edges list of edge objects
#' @param enableFileInputs logical
#' @param width widget width
#' @param height widget height
#'
#' @export
layoutWidget <- function(nodes = list(),
                         edges = list(),
                         enableFileInputs = FALSE,
                         width = NULL,
                         height = NULL) {

  htmlwidgets::createWidget(
    name = "layoutWidget",
    x = list(
      nodes = nodes,
      edges = edges,
      enableFileInputs = enableFileInputs
    ),
    width = width,
    height = height,
    package = "DAGnabit",

    # ✅ THIS IS THE MISSING PIECE
    sizingPolicy = htmlwidgets::sizingPolicy(
      defaultWidth  = "100%",
      defaultHeight = 600,
      viewer.defaultHeight = 600,
      viewer.fill = TRUE
    )
  )
}

File: inst/htmlwidgets/layoutWidget.js
HTMLWidgets.widget({

  name: 'layoutWidget',
  type: 'output',

  factory: function(el, width, height) {

    let editor = null;

    return {

      renderValue: function(x) {

        el.innerHTML = "";

        // ✅ CRITICAL: force real pixel size for RStudio Viewer
        el.style.width  = width  + "px";
        el.style.height = height + "px";

        editor = window.initLayoutEditor(el, {
          nodes: x.nodes || [],
          edges: x.edges || [],
          enableFileInputs: x.enableFileInputs ?? false,
          fitOnDataLoad: true,
          width: width,
          height: height
        });

      },

      resize: function(width, height) {
        if (editor && editor.resize) {
          editor.resize();
        }
      }
    };
  }
});
